add functionality to dynamically store viewport dimensions

BurningTreeC opened the pull request:
Hi, as discussed [here](https://groups.google.com/forum/#!topic/tiddlywikidev/MUHKAUG-0iY) I've added my version of an implementation to store the viewport dimensions (if enabled) in $:/state/viewport or a user-specified tiddler

Simon

Jermolene commented:
Great stuff, thank you @BurningTreeC. I've left some minor comments, let me know if anything doesn't make sense.

BurningTreeC commented:
thanks @Jermolene ,
I couldn't find your comments, maybe I'm missing something or they're not yet there,

 I'll come back later!

Jermolene commented:
Hi @BurningTreeC let me know if you can't see the comments; they are showing up here for me

BurningTreeC commented:
no, nothing, I'm sorry

I'm not that experienced with github-commiting - possibly I did something wrong
There are no comments visible on my end, or they're perfectly hidden

BurningTreeC commented on plugins/tiddlywiki/dynaview/dynaview.js:
```diff
@@ -18,7 +18,15 @@ exports.platforms = ["browser"];
+
+if($tw.wiki.tiddlerExists("$:/config/dynaview")) {
+	var configTiddler = $tw.wiki.getTiddler("$:/config/dynaview");
+	exportViewportDimensions = configTiddler.fields["viewport-dimensions"] === "yes";
```
no, only keeping the number of created tiddlers low, if there will be more configurations at some point. I was in doubt and therefor did it like this. I'll change it to use text fields

BurningTreeC commented on plugins/tiddlywiki/dynaview/dynaview.js:
```diff
@@ -34,6 +42,11 @@ function onScrollOrResize(event) {
+			if(exportViewportDimensions) {
+				saveViewportDimensions();
+			} else if($tw.wiki.tiddlerExists(viewportTiddler)) {
+				$tw.wiki.deleteTiddler(viewportTiddler);
```
my logic - not having wrong values somewhere which can cause strange behaviour - if one forgets that there might still be those values used somewhere inside a macro or similar.

This is more a savior for myself, I often forget about such things.

I'll remove it - I get your point - and I'll take more care in my future wikies 👍

BurningTreeC commented on plugins/tiddlywiki/dynaview/dynaview.js:
```diff
@@ -18,7 +18,15 @@ exports.platforms = ["browser"];
+    exportViewportDimensions = false,
+    viewportTiddler;
+
+if($tw.wiki.tiddlerExists("$:/config/dynaview")) {
```
I didn't think about it - I'll move it inside the startup() function

BurningTreeC commented on plugins/tiddlywiki/dynaview/dynaview.js:
```diff
@@ -18,7 +18,15 @@ exports.platforms = ["browser"];
+if($tw.wiki.tiddlerExists("$:/config/dynaview")) {
+	var configTiddler = $tw.wiki.getTiddler("$:/config/dynaview");
+	exportViewportDimensions = configTiddler.fields["viewport-dimensions"] === "yes";
+	viewportTiddler = configTiddler.fields["viewport-tiddler"] || "$:/state/viewport";
```
it isn't worth anything - I'll simply set it to $:/state/viewport if that title is ok

BurningTreeC commented on plugins/tiddlywiki/dynaview/dynaview.js:
```diff
@@ -91,4 +104,11 @@ function checkVisibility() {
+function saveViewportDimensions() {
+	var viewportWidth = window.innerWidth || document.documentElement.clientWidth,
+	    viewportHeight = window.innerHeight || document.documentElement.clientHeight;
+	$tw.wiki.setText(viewportTiddler,"viewport-width",undefined,viewportWidth, { suppressTimestamp: true });
```
I initially didn't really think about which type of tiddler to use, had just a system tiddler and that caused the saving button to shine red far too often. saving a wiki brings up the bottom "saving bar" in chromium, which changes the viewport and triggers a resize.

It's very interesting for me to think more specifically about these little tw details, haven't done that before

I'll change the options to undefined?

BurningTreeC commented on plugins/tiddlywiki/dynaview/config.tid:
```diff
@@ -0,0 +1,8 @@
+title: $:/plugins/tiddlywiki/dynaview/config
+
+//These settings take effect after reloading your wiki//
```
I'll investigate the language part for english, german and italian

Inside onScrollOrResize I need another check if it's enabled - then it should be possible to enable it dynamically
A check on startup then only makes sense if the dimensions get saved at startup - does the loading of the page already emit a resize event?

BurningTreeC commented:
Hi Jeremy,
I think I've addressed all issues (but the language part)
It stores the values at startup, on every resize and it can dynamically be switched on/off
It uses the text field of $:/config/ViewportDimensions for enabling/disabling...
...and stores the values in the text fields of two state tiddlers: $:/state/viewport/width $:/state/viewport/height

About the lingo part - I don't know how to handle that part, or how to do that from within a plugin, where should the language tiddlers be placed?

Simon

Jermolene commented:
Hi @BurningTreeC 

> About the lingo part - I don't know how to handle that part, or how to do that from within a plugin, where should the language tiddlers be placed?

My apologies, entirely my mistake: I briefly forgot that this was for a plugin. At the moment we don't translate the text in the core plugins.

Jermolene commented:
Thanks @BurningTreeC
